easybuild/tools/github.py

```diff
@@ -408,6 +408,56 @@ def post_comment_in_issue(issue, txt, repo=GITHUB_EASYCONFIGS_REPO, github_user=
         raise EasyBuildError("Failed to create comment in PR %s#%d; status %s, data: %s", repo, issue, status, data)
 
 
+def setup_repo(git_repo, github_url, target_account, branch_name):
+    """
+    Set up repository by checking out specified branch from repo at specified URL.
+
+    @param git_repo: git.Repo instance
+    @param github_url: URL to GitHub repo
+    @param target_account: name of GitHub account that owns GitHub repo
+    @param branch_name: name of branch to check out
+    """
+    _log.debug("Cloning from %s", github_url)
+
+    # salt to use for names of remotes/branches that are created
+    salt = ''.join(random.choice(string.letters) for _ in range(5))
+
+    remote_name = 'pr_target_account_%s_%s' % (target_account, salt)
+
+    origin = git_repo.create_remote(remote_name, github_url)
+    if not origin.exists():
+        raise EasyBuildError("%s does not exist?", github_url)
+
+    # git fetch
+    # can't use --depth to only fetch a shallow copy, since pushing to another repo from a shallow copy doesn't work
+    print_msg("fetching branch '%s' from %s..." % (branch_name, github_url))
+    try:
+        res = origin.fetch()
+    except GitCommandError as err:
+        raise EasyBuildError("Failed to fetch branch '%s' from %s: %s", branch_name, github_url, err)
+    if res:
+        if res[0].flags & res[0].ERROR:
+            raise EasyBuildError("Fetching branch '%s' from remote %s failed: %s", branch_name, origin, res[0].note)
+        else:
+            _log.debug("Fetched branch '%s' from remote %s (note: %s)", branch_name, origin, res[0].note)
+    else:
+        raise EasyBuildError("Fetching branch '%s' from remote %s failed: empty result", branch_name, origin)
+
+    # git checkout -b <branch>; git pull
+    if hasattr(origin.refs, branch_name):
+        origin_branch = getattr(origin.refs, branch_name)
+    else:
+        raise EasyBuildError("Branch '%s' not found at %s", branch_name, github_url)
+
+    _log.debug("Checking out branch '%s' from remote %s", branch_name, github_url)
+    try:
+        origin_branch.checkout(b=branch_name)
+    except GitCommandError as err:
+        alt_branch = 'pr_start_branch_%s_%s' % (branch_name, salt)
+        _log.debug("Trying to work around checkout error ('%s') by using different branch name '%s'", err, alt_branch)
+        origin_branch.checkout(b=alt_branch)
```

Comment:
I can wrap this into a try/except too, but if this fails, there's nothing much we can do except throw an exception, because then something is very wrong/weird.